docs/ref/models/expressions.txt

```diff
@@ -588,22 +588,40 @@ expressions::
 
 Next, we write the method responsible for generating the SQL::
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
         sql_expressions, sql_params = [], []
         for expression in self.expressions:
             sql, params = compiler.compile(expression)
             sql_expressions.append(sql)
             sql_params.extend(params)
-        self.extra['expressions'] = ','.join(sql_expressions)
-        return self.template % self.extra, sql_params
+        data = self.extra.copy()
+        data.update(**kwargs)
+        template = data.get('template', self.template)
+        data['expressions'] = ','.join(sql_expressions)
+        return template % data, params
 
-    def as_oracle(self, compiler, connection):
+    def as_oracle(self, compiler, connection, **kwargs):
         """
         Example of vendor specific handling (Oracle in this case).
         Let's make the function name lowercase.
         """
-        self.template = 'coalesce( %(expressions)s )'
-        return self.as_sql(compiler, connection)
+        return self.as_sql(compiler, connection,
+            template='coalesce( %(expressions)s )'
+        )
+
+.. versionchanged:: 1.10
+
+    ``as_sql``-methods now support ``kwargs`` to override the default
```

Comment:
Try to make the note "self-contained" as described in https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#documenting-new-features.